src/test/java/specs/JUnitRuleExample.java

```diff
@@ -0,0 +1,90 @@
+package specs;
+
+import static com.greghaskins.spectrum.Spectrum.applyRules;
+import static com.greghaskins.spectrum.Spectrum.applyRulesHere;
+import static com.greghaskins.spectrum.Spectrum.describe;
+import static com.greghaskins.spectrum.Spectrum.it;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+import com.greghaskins.spectrum.Spectrum;
+
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+@RunWith(Spectrum.class)
+public class JUnitRuleExample {
+  public static class TempFolderRule {
+    @Rule
+    public TemporaryFolder tempFolderRule = new TemporaryFolder();
+  }
+
+  {
+    final Set<File> filesSeen = new HashSet<>();
+    describe("A parent spec", () -> {
```

Comment:
This suite was a little hard to follow the first couple times through reading it. Maybe the assertions could be more direct, something like `assertTrue(theTempFolder.exists())` and/or  `assertThat(filesSeen, not(hasItem(theTempFolder)))`.